### ISSUE TITLE
Add support for LVMv2

### ISSUE THREAD

**Issue by heylix** [archlinux/archinstall#584]
Hello, I joined the Matrix channel and asked if any of my LUKS-on-LVM or LVM-on-LUKS setups were possible and we found out they're indeed, but not with archinstall.

I request addition of LVMv2 support, both for LVM-on-LUKS (so have a LVM VG encrypted with LUKS) and LUKS-on-LVM (have a LVM VG which is unencrypted per se, but you can have LVs which you can encrypt with LUKS).

#105 already discussed this but in a broader sense.

My suggestion would be, since this is a fairly advanced use case, to only support this when keeping the existing layout to simplify things. The installer could then ask for the existing password and unlock the LVM partitions it needs to access to install the system.

**Comment by Torxed** [comment]
I agree that the easiest way to support this would be to use `/mnt` as is, and let the installer detect the necessary kernel parameters (if any are needed).

**Comment by Torxed** [comment]
The other day I re-wrote a large portion of the partitioning logic, my goal is to start on [btrfs subvolumes](https://github.com/archlinux/archinstall/issues/93) after that's stable enough and then lvm. Getting this right will probably take some time - as I don't want to be left with something that doesn't work with or without encryption for instance. But the goal is to support the most common partition/disk setups including LVM : )

**Comment by svartkanin** [comment]
@Torxed i will start working on LVM support. Is there anything you had in mind for it already?
I was thinking something similar to the partitioning menu atm, a new entry of default lvm layout and a custom option.
That will then also require neq encryption options such as lvmOnLuks LuksOnLvm

**Comment by Torxed** [comment]
> I was thinking something similar to the partitioning menu atm

I think you'll solve that much neater than me, so I'll give you free reign to cook something up! :)

**Comment by svartkanin** [comment]
That sounds like expectations are high 😆
